Auto generate friendly name from scaffold

synmal opened the pull request:
This will auto generate readable i18n from preside scaffold.

For example, if an object is created from Preside terminal with fields of:
1. "name"
2. "full_name"
3. "address_1"
4. "address_2"
5. "very_very_long_address_field"

It will generate as follows:
1. "Name"
2. "Full Name"
3. "Address 1"
4. "Address 2"
5. "Very Very Long Address Field"

maljer committed:
Auto generate friendly name from scaffold

sebduggan commented:
Nice enhancement, Malcolm!

May I suggest an amendment to the convert function?

```
function _convertToReadableFormat( required string words ){
	var removedNonChar = reReplace( arguments.words, "[^a-zA-Z0-9]+", " ", "all" );	
	return reReplaceNocase( removedNonChar, "\b(\w)", "\u\1", "all" );
}
```

First we add the "+" to condense multiple non-characters into a single space. But then we replace the whole of the word array looping with a simple regular expression, which is probably quicker and is much neater!

maljer committed:
Minor amendment in the _convertToReadableFormat function by Seb.

synmal commented:
Hi Seb,

Thanks for the suggestion. The amendments you requested already been pushed.

DominicWatson commented:
Super super l33t. Merging into upcoming 10.11.0 release as an enhancement.

DominicWatson commented on system/services/devtools/ScaffoldingService.cfc:
```diff
@@ -81,13 +81,13 @@ component singleton=true {
 
-		i18nProps[ "name" ]        = arguments.name;
-		i18nProps[ "description" ] = arguments.description;
+		i18nProps[ "name" ]        = _convertToReadableFormat( words = arguments.name );
```
I think we should only be converting inputs that are field names. Descriptions and name's are already input by users as free text, not constrained. Places I believe we should convert:

1. Field names in preside objects
2. Field names in system config forms (`new configform` command)
3. Field names in widget config forms (`new widget` command)
4. Field names in page types (`new pagetype` command)

synmal replied:
Nice catch there. Totally overlooked the names and descriptions as free text. Changes now committed and pushed.

DominicWatson reviewed (changes requested):
Hey @synmal , love this PR. Could you take a look at the review feedback re which inputs we should be converting. I'll then merge into upcoming 10.11 release.

maljer committed:
Merge remote-tracking branch 'upstream/stable' into PRESIDECMS-1313_auto_generate_readable_i18n

maljer committed:
Removed _convertToReadableFormat() from arguments that are already in as free text.
The ones that will be converted to readable format are:
1. Field names when scaffolding objects.
2. Field names when scaffolding system config form.
3. Field names when scaffolding widget config form.
4. Field names when scaffolding page-types.

DominicWatson commented:
Nice one, thanks @synmal :) This has now been merged into the upcoming release 10.11.0. This issue will auto close when the release is finalized.
